File: src/main/java/com/otaviolarrosa/identitymanager/application/user/create/CreateUserService.java
package com.otaviolarrosa.identitymanager.application.user.create;

import java.util.UUID;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.otaviolarrosa.identitymanager.infrastructure.database.entities.UserEntity;
import com.otaviolarrosa.identitymanager.infrastructure.database.repositories.UserRepository;
import com.otaviolarrosa.identitymanager.infrastructure.encryption.Encryption;

@Service
public class CreateUserService {
	
	@Autowired
	private CreateUserValidator validator;
	
	@Autowired
	private UserRepository userRepository;
	
	@Autowired
	private Encryption encryptor;
	
	public CreateUserResult handleExecution(CreateUserInput createUserInput) {
		CreateUserResult result = new CreateUserResult();
		result = validator.validate(createUserInput);
		if(!result.isValid()) {
			return result;
		}
		
		UserEntity entity = new UserEntity();
		entity.setEmail(createUserInput.getEmail());
		entity.setFirstName(createUserInput.getFirstName());
		entity.setLastName(createUserInput.getLastName());
				
		String password = encryptor.encrypt(createUserInput.getPassword());
		entity.setPassword(password);
		
		UUID userCode = java.util.UUID.randomUUID();
		entity.setUserCode(userCode);
		
		userRepository.save(entity);
		result.setUserCode(userCode);
		return result;
	}
}

File: src/main/kotlin/com/otaviolarrosa/singleSignOn/application/user/create/CreateUserService.kt
package com.otaviolarrosa.singleSignOn.application.user.create

import com.otaviolarrosa.singleSignOn.infrastructure.database.entities.UserEntity
import com.otaviolarrosa.singleSignOn.infrastructure.database.repositories.UserRepository
import com.otaviolarrosa.singleSignOn.infrastructure.database.security.Encrytptor
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class CreateUserService @Autowired constructor(
    private val userRepository: UserRepository,
    private val validator: CreateUserValidator,
    private val encrytptor: Encrytptor
){

    fun handleExecution(input: CreateUserInput) : CreateUserResult {
        val result: CreateUserResult = validator.validate(input)

        if(!result.valid){
            return result
        }

        val entity = UserEntity()
        entity.firstName = input.firstName
        entity.lastName = input.lastName
        entity.email = input.email
        entity.password =  encrytptor.encrypt(input.password)
        val userCode = UUID.randomUUID()
        entity.userCode = userCode
        result.userCode = userCode

        userRepository.save(entity)
        return result
    }
}